airbyte-integrations/connectors/source-marketo/source_marketo/source.py

```diff
@@ -0,0 +1,530 @@
+                elif attr["dataType"] in ["float", "currency"]:
+                    field_schema = {"type": "number"}
+                elif attr["dataType"] == "boolean":
+                    field_schema = {"type": "boolean"}
+                elif attr["dataType"] in STRING_TYPES:
+                    field_schema = {"type": "string"}
+                elif attr["dataType"] in ["array"]:
+                    field_schema = {"type": "array", "items": {"type": ["integer", "number", "string", "null"]}}
+                else:
+                    field_schema = {"type": "string"}
+
+                field_schema["type"] = [field_schema["type"], "null"]
+
+                properties[attr_name] = field_schema
+
+        schema = {
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "type": ["null", "object"],
+            "additionalProperties": False,
+            "properties": properties,
+        }
+
+        return schema
+
+
+class ActivityTypes(MarketoStream):
+    """
+    Return list of all activity types.
+    API Docs: http://developers.marketo.com/rest-api/lead-database/activities/#describe
+    """
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "/rest/v1/activities/types.json"
+
+
+class Programs(IncrementalMarketoStream):
+    """
+    Return list of all programs.
+    API Docs: http://developers.marketo.com/rest-api/assets/programs/#by_date_range
+    """
+
+    cursor_field = "updatedAt"
+    page_size = 200
+    offset = 0
+
+    def path(self, **kwargs) -> str:
+        return f"/rest/asset/v1/{self.name}.json"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        data = response.json().get(self.data_field)
+
+        if data:
+            self.offset += self.page_size + 1
+            return {"offset": self.offset}
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        """
+        Programs are queryable via their updatedAt time but require and
+        end date as well. As there is no max time range for the query,
+        query from the bookmark value until current.
+        """
+
+        params = super().request_params(stream_state, stream_slice, next_page_token)
+        params.update(
+            {
+                "maxReturn": self.page_size,
+                "earliestUpdatedAt": stream_slice["startAt"],
+                "latestUpdatedAt": stream_slice["endAt"],
+            }
+        )
+
+        return params
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, stream_state, **kwargs):
+            record["updatedAt"] = record["updatedAt"][:-5]
+            yield record
```

Comment:
Do you try to remove a time part?
Please add a description about this. Or use some python lib for datetime conversion.